pkg/node/fsm/ng_state.go

```diff
@@ -173,28 +230,29 @@ func (a *NGState) MinedBlock(
 }
 
 func (a *NGState) MicroBlock(p peer.Peer, micro *proto.MicroBlock) (State, Async, error) {
-	metrics.FSMMicroBlockReceived("ng", micro, p.Handshake().NodeName)
-	block, err := a.checkAndAppendMicroblock(micro) // the TopBlock() is used here
-	if err != nil {
-		metrics.FSMMicroBlockDeclined("ng", micro, err)
-		return a, nil, a.Errorf(err)
+	if a.microBlockWaitingForSnapshot != nil {
+		return a, nil, a.Errorf(errors.Errorf(
+			"skip micro block %s, waiting snapshot for micro block %s",
+			micro.TotalBlockID.String(),
+			a.microBlockWaitingForSnapshot.TotalBlockID.String()))
 	}
-	zap.S().Named(logging.FSMNamespace).Debugf(
-		"[%s] Received microblock '%s' (referencing '%s') successfully applied to state",
-		a, block.BlockID(), micro.Reference,
-	)
-	a.baseInfo.MicroBlockCache.Add(block.BlockID(), micro)
-	a.blocksCache.AddBlockState(block)
-	a.baseInfo.scheduler.Reschedule()
-
-	// Notify all connected peers about new microblock, send them microblock inv network message
-	if inv, ok := a.baseInfo.MicroBlockInvCache.Get(block.BlockID()); ok {
-		//TODO: We have to exclude from recipients peers that already have this microblock
-		if err = a.broadcastMicroBlockInv(inv); err != nil {
-			return a, nil, a.Errorf(errors.Wrap(err, "failed to handle microblock message"))
+	metrics.FSMMicroBlockReceived("ng", micro, p.Handshake().NodeName)
+	if !a.baseInfo.enableLightMode {
+		block, err := a.checkAndAppendMicroBlock(micro, nil) // the TopBlock() is used here
+		if err != nil {
+			metrics.FSMMicroBlockDeclined("ng", micro, err)
+			return a, nil, a.Errorf(err)
 		}
+		zap.S().Named(logging.FSMNamespace).Debugf(
+			"[%s] Received microblock '%s' (referencing '%s') successfully applied to state",
+			a, block.BlockID(), micro.Reference,
+		)
+		a.baseInfo.MicroBlockCache.AddMicroBlockWithSnapshot(block.BlockID(), micro, &proto.BlockSnapshot{})
```

Comment:
Why `snapshot` argument for `AddMicroBlockWithSnapshot` is not `nil`? Is it intended?